lib/rollout.rb

```diff
@@ -159,6 +166,33 @@ def define_group(group, &block)
     @groups[group.to_sym] = block
   end
 
+  def define_id_collection(name, ids = [])
+    @storage.set("collection:#{name}", ids.join(","))
+    @storage.set(collection_key, (collections.keys | [name.to_sym]).join(","))
+  end
+
+  def add_collection_to_feature(collection, feature)
+    with_feature(feature) do |feature|
+      @storage.set(collection_feature_key(feature.name), (feature_collections(feature.name).keys | [ collection.to_sym]).join(","))
+    end
+  end
+
+  def collections
+    Hash[*retrieve_from_storage(collection_key).map{ |collect| [ collect.to_sym, @storage.get("collection:#{collect}") ] }.flatten]
```

Comment:
im trying to understand what this does:

```
retrieve_from_storage(collection_key = "__collections__")
=> Returns array [1,2,3]
map it to each collection
=> Returns nested array [ [1, [4]], [2, [6]], [3, [7]] ] => flatten [1,4,2,6,3,7]
Hash[*returned array] = Hash[ 1, 4, 2, 6, 3, 7 ]
=> {1=>4, 2=>6, 3=>7}
```
I can't really understand what you are doing here, could u explain a bit? (also, you could use ```collection_key(collect)```)